xbmc/windowing/osx/WinEventsOSXImpl.mm

```diff
@@ -341,4 +321,28 @@ - (NSEvent*)InputEventHandler:(NSEvent*)nsevent
     return nullptr;
 }
 
+- (XBMC_Event)keyPressEvent:(CGEventRef*)event
+{
+  UniChar unicodeString[MAXUNICODESTRLEN] = {0};
+  UniCharCount actualStringLength = 0;
+
+  CGKeyCode keycode = (CGKeyCode)CGEventGetIntegerValueField(*event, kCGKeyboardEventKeycode);
+  CGEventKeyboardGetUnicodeString(*event, MAXUNICODESTRLEN,
```

Comment:
instead of hardcoding max amount it's probably better to make 2 calls to `CGEventKeyboardGetUnicodeString` where first would return the real length. From the header:

> When you call this function with a NULL string or a maximum string length of 0, the function still returns the actual count of Unicode characters in the event.

Although we care only about the first character, then even length 2 should be sufficient.